### ISSUE TITLE
tensorflow.python.keras.engine.functional.Functional Exception: Data must be 1-dimensional

### ISSUE THREAD

**Issue by pa-blan** [ModelOriented/DALEX#367]
Hi,
I want to use Dalex with a model created from Autkeras, with a self-made Sequential Model it worked. What is the problem?

Model from Autokeras structured_data_classifier : <tensorflow.python.keras.engine.functional.Functional at 0x1bbab227340>
https://autokeras.com/structured_data_classifier/

```
explainer = dx.Explainer(
    model = model,
    data = X_test,
    y = y_test,
    model_type= "classification")

#%%
explainer.model_performance()

explainer.model_parts().plot()
```

**Comment by hbaniecki** [comment]
Can you provide a minimal code example to reproduce this behaviour?

**Comment by pa-blan** [comment]
import autokeras as ak 

clf = ak.StructuredDataClassifier(
    max_trials=10, # It tries 5 different models.
    overwrite=True,
    metrics = [
        # f1_score,
        # tf.keras.metrics.TruePositives(name='tp'),
        # tf.keras.metrics.FalsePositives(name='fp'),
        # tf.keras.metrics.TrueNegatives(name='tn'),
        # tf.keras.metrics.FalseNegatives(name='fn'), 
        tf.keras.metrics.BinaryAccuracy(name='accuracy'),
        tf.keras.metrics.Precision(name='precision'),
        tf.keras.metrics.Recall(name='recall'),
        # tf.keras.metrics.AUC(name='auc')
    ],
    num_classes  = 2,
    objective = kt.Objective('val_recall', direction='max'), #kt.Objective("accuracy", direction="max"),
    loss = tf.keras.losses.BinaryCrossentropy(from_logits = True),
    tuner = "random",
    seed = 42,
    project_name = "AUTOKERAS_TEST2",

)
#%%
import numpy as np
from sklearn.utils import class_weight
class_weight = class_weight.compute_class_weight('balanced',np.unique(y_train),y_train)
class_weight = dict(enumerate(class_weight))
class_weight

#%%
# Feed the structured data classifier with training data.
clf.fit(X_train,
        y_train,
        # Split the training data and use the last 15% as validation data.
        validation_split=0.2,
        class_weight = class_weight,
        epochs=10)


#%%
clf.predict(X_test)[1][0]


#%%
# Predict with the best model.
# predicted_y = clf.predict(X_test)
# Evaluate the best model with testing data.
print(clf.evaluate(X_test, y_test))

#%%

model = clf.export_model()
model.summary()

#%%
model
#%%
model.predict(X_test)[1]

#%%
import dalex as dx
dx.__version__

#%%

explainer = dx.Explainer(
    model = model,
    data = X_test,
    y = y_test,
    model_type= "classification")

#%%
explainer.model_performance()

#%%
explainer.model_parts().plot()

#%%
explainer.model_parts().plot()

**Comment by hbaniecki** [comment]
What is `kt`: `kt.Objective('val_recall', direction='max')`

**Comment by pa-blan** [comment]
Keras-tuner

**Comment by pa-blan** [comment]
@hbaniecki it worked with the Regression type in the dx. Explainer, but its a Classifier, strange

explainer = dx.Explainer(
    model = model,
    data = X_test,
    y = y_test,
    model_type= "Regression")

**Comment by hbaniecki** [comment]
@pa-blan Thanks for this issue!

I have basically added `tensorflow.python.keras.engine.functional.Functional` to the list, although by default it will detect such model as `'regression'` (because it returns one column) so `model_type='classification'` is needed. 

I wish that `autokeras` implemented `predict_proba` so that `dalex` can be smarter.

You can install the dev version and test it using
`pip install git+https://git@github.com/ModelOriented/DALEX.git@master#subdirectory=python/dalex`
Example: [autokeras.zip](https://github.com/ModelOriented/DALEX/files/5624947/autokeras.zip)

**Comment by pbiecek** [comment]
Thanks for such a quick solution!

**Comment by pa-blan** [comment]
@hbaniecki cant install the dev version, does not go any further.

(test) PS C:\Users\i00110392> pip install git+https://git@github.com/ModelOriented/DALEX.git@master#subdirectory=python/dalex
Collecting git+https://****@github.com/ModelOriented/DALEX.git@master#subdirectory=python/dalex
  Cloning https://****@github.com/ModelOriented/DALEX.git (to revision master) to c:\users\i00110392\appdata\local\temp\pip-req-build-wl4wlsq8

**Comment by hbaniecki** [comment]
For me clicking `enter` helped.

**Comment by pa-blan** [comment]
Is this a large package?

**Comment by hbaniecki** [comment]
@pa-blan How about `pip install git+https://git@github.com/ModelOriented/DALEX.git@master#subdirectory=python/dalex --no-input`?

**Comment by pa-blan** [comment]
@pbiecek  no doesn't work

**Comment by hbaniecki** [comment]
Weird, `pip install git+https://github.com/slundberg/shap --no-input` works for me the same.

 You can clone the whole repo (master) -> get into the `./python/dalex` folder -> use `pip install .`

**Comment by pa-blan** [comment]
Now it has worked. Thank you very much for the very fast help. I hope Dalex gets more attention in the world. I am looking forward to the implementation of Pycaret and Catboost. :-)

**Comment by hbaniecki** [comment]
@pa-blan what is your idea for `catboost`? It works with `dalex` through `numpy` and `pandas`. Do you mean some support for `Pool`?

**Comment by pa-blan** [comment]
@pbiecek  oh my fault, I don't tried it yet with Catboost.
